vendor/bat-native-ledger/src/bat/ledger/internal/uphold/uphold_authorization_unittest.cc

```diff
@@ -0,0 +1,299 @@
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include <tuple>
+#include <utility>
+
+#include "base/test/task_environment.h"
+#include "bat/ledger/internal/database/database_mock.h"
+#include "bat/ledger/internal/ledger_client_mock.h"
+#include "bat/ledger/internal/ledger_impl_mock.h"
+#include "bat/ledger/internal/state/state_keys.h"
+#include "bat/ledger/internal/uphold/uphold_authorization.h"
+#include "bat/ledger/option_keys.h"
+#include "net/http/http_status_code.h"
+
+// npm run test -- brave_unit_tests --filter=UpholdAuthorizationTest*
+
+using ::testing::_;
+using ::testing::Invoke;
+using ::testing::TestParamInfo;
+using ::testing::TestWithParam;
+using ::testing::Values;
+
+namespace ledger {
+namespace uphold {
+namespace tests {
+
+using GetAnonFundsParamType =
+    std::tuple<bool,               // fetch old balance enabled
+               std::string,        // Brave wallet
+               type::UrlResponse,  // balance server response
+               std::pair<type::Result,
+                         base::Optional<double>>  // expected result
+               >;
+
+class GetAnonFunds : public TestWithParam<GetAnonFundsParamType> {
+ private:
+  base::test::TaskEnvironment scoped_task_environment_;
+
+ protected:
+  std::unique_ptr<MockLedgerClient> mock_ledger_client_;
+  std::unique_ptr<MockLedgerImpl> mock_ledger_impl_;
+  std::unique_ptr<database::MockDatabase> mock_database_;
+  std::unique_ptr<UpholdAuthorization> authorization_;
+
+ public:
+  GetAnonFunds()
+      : mock_ledger_client_{std::make_unique<MockLedgerClient>()},
+        mock_ledger_impl_{
+            std::make_unique<MockLedgerImpl>(mock_ledger_client_.get())},
+        mock_database_{
+            std::make_unique<database::MockDatabase>(mock_ledger_impl_.get())},
+        authorization_{
+            std::make_unique<UpholdAuthorization>(mock_ledger_impl_.get())} {}
+};
+
+std::string GetAnonFundsNameSuffixGenerator(
+    const TestParamInfo<GetAnonFunds::ParamType>& info) {
+  const bool fetch_old_balance_enabled = std::get<0>(info.param);
+  const std::string& brave_wallet = std::get<1>(info.param);
+  const type::UrlResponse& balance_server_response = std::get<2>(info.param);
+
+  if (!fetch_old_balance_enabled) {
+    return "fetch_old_balance_disabled";
+  }
+
+  if (brave_wallet.empty()) {
+    return "brave_wallet_is_not_created";
+  }
+
+  if (brave_wallet.find(R"("payment_id": "")") != std::string::npos) {
+    return "brave_wallet_payment_id_is_empty";
+  }
+
+  if (balance_server_response.status_code != net::HttpStatusCode::HTTP_OK) {
+    return "balance_server_error";
+  }
+
+  if (balance_server_response.body.empty()) {
+    return "invalid_body_in_balance_server_response";
+  }
+
+  return "happy_path";
+}
+
+INSTANTIATE_TEST_SUITE_P(
+    UpholdAuthorizationTest,
+    GetAnonFunds,

```

Comment:
Nit: I feel like the first param should be `UpholdAuthorization` and the second param should be `GetAnonFundsTest` just for consistency with how other tests are written in Chromium.

Reply:
Most of the tests I saw had a `Test` postfix in their names (e.g. `TEST_F(PostClaimUpholdTest, ServerError400) { /* ... */ }`, `TEST_F(PostTransactionUpholdTest, ServerError500) { /* ... */ }`, `TEST_F(UpholdTest, FetchBalanceConnectedWallet) { /* ... */ }`, etc.), hence the name `UpholdAuthorizationTest`.
But I like your idea better! Which one should we prefer? :grin:

Reply:
See my other comment below... :-)